internal/catalogmetadata/sort/sort.go

```diff
@@ -0,0 +1,34 @@
+package sort
+
+import (
+	"github.com/operator-framework/operator-controller/internal/catalogmetadata"
+)
+
+// ByVersion is a sort "less" function that orders bundles
+// in inverse version order (higher versions on top).
+func ByVersion(b1, b2 *catalogmetadata.Bundle) bool {
+	ver1, err1 := b1.Version()
+	ver2, err2 := b2.Version()
+	errComp := compareErrors(err1, err2)
+	if errComp != 0 {
+		return errComp < 0
+	}
```

Comment:
If both versions come back nil and with errors, then line 19 will panic:
```suggestion
	errComp := compareErrors(err1, err2)
	if errComp != 0 {
		return errComp < 0
	} else if ver1 == nil && ver2 == nil {
		return 0
	}
```
If only one version comes back nil then the `errComp` check should catch it.

Reply:
Good catch! I'll update tests and code.

Your suggested code won't work since we need to return bool (true if `b1` is less than `b2` and false otherwise). But we can do something like this:

```go
if err1 != nil || err2 != nil {
	return compareErrors(err1, err2) < 0
}
```